bookshelter-app/src/components/SearchInput/index.tsx

```diff
@@ -1,15 +1,65 @@
 import styles from './index.module.css';
-
+import { useState, useEffect } from 'react';
 import Glass from '../Icons/Glass';
-const SearchInput = () => {
+import { get } from '../../services/api';
+import { API } from '../../constants/api';
+import { Book } from '../../types';
+import { useNavigate } from 'react-router-dom';
+
+interface SearchInputProps {
+  setResults: (results: Book[]) => void;
+  onOpen: () => void;
+  onclose: () => void;
+}
+
+const SearchInput = ({ setResults, onOpen, onclose }: SearchInputProps) => {
+  const [searchTerm, setSearchTerm] = useState('');
+  const navigate = useNavigate();
+
+  useEffect(() => {
```

Comment:
don't handle logic in component, let's move it to parent component

Reply:
Fixed!